Add fedora 33 / 34 support

ptoal opened the pull request:
<!--- Provide a short summary of your changes in the Title above -->

Minor changes to enable this role to work on Fedora 33/34 targets

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:
- [X] I have read the **CONTRIBUTING** document.
- [X] I have run the pre-merge tests locally and they pass.
- [X] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [] All new and existing tests passed.

ptoal committed:
Add fedora 33 / 34 support

jmidgren commented on tasks/set_facts.yml:
```diff
@@ -22,6 +22,14 @@
+    kea_dhcp_pre_reqs:
+      - dnf-utils
+  when:
+    - ansible_os_family == "Fedora"
```
```suggestion
    - ansible_distribution == "Fedora"
```

jmidgren replied:
I assume it to should be `ansible_distribution == "Fedora"`, perhaps even complemented by `ansible_os_family == "RedHat"`

jmidgren reviewed (changes requested):
How nice to see that someone else took the time to add support for Fedora 34 (and 33). I just installed the role from Ansible Galaxy and found out myself that it did not work in Fedora 34. Thank you!

One additional thing to consider might be to add a comment in `defaults/main.yml` at the `kea_dhcp_version` variable to let users know that installs on Fedora will not be affected by this variable.

ptoal commented:
I don't know how I missed that.  I think I must have made a mistake when I was re-basing, and deleted a commit before I pushed.  I have corrected that.

> One additional thing to consider might be to add a comment in `defaults/main.yml` at the `kea_dhcp_version` variable to let users know that installs on Fedora will not be affected by this variable.

I didn't bother with different versions, because I didn't need anything other than the default included in the fedora repos.  I notice that the kea_dhcp_version doesn't have full effect for all distros, so I'm not sure if a fedora specific comment is best, or if a general comment that there is a limited selection depending on distro.

jmidgren commented:
> > One additional thing to consider might be to add a comment in `defaults/main.yml` at the `kea_dhcp_version` variable to let users know that installs on Fedora will not be affected by this variable.
> 
> I didn't bother with different versions, because I didn't need anything other than the default included in the fedora repos. I notice that the kea_dhcp_version doesn't have full effect for all distros, so I'm not sure if a fedora specific comment is best, or if a general comment that there is a limited selection depending on distro.

I looked into ISC's repository of built packages for various distributions and it seems like they build kea 1.9 for Fedora 33 and 34. Older still maintained releases (e.g 1.8) is only built for Fedora 31 and 32 (and perhaps not at all any longer since those Fedora release are EOL). Maybe it's a bad idea to prevent installation from ISC's repositories for Fedora although it is available in Fedora's own repos? On the other hand it could be that Fedora will always ship with the current development version of kea, perhaps even the exact same version that is available in ISC's repos?

Anyway, one way to add flexibility would be to add a hard-coded value to the `kea_dhcp_repos` variable, e.g. `distribution` or `distro`. This would mean that packages should be installed from the distribution's own repo instead of from a provided URL. It would work also for other distros as they add kea to their repos over time. An assert would fire if the package is not in the repo for current distro. Is this a good idea or does is add unnecessary complexity? Perhaps it should be its own variable? I can provide a suggestion if you like the idea.

There is [an other PR](https://github.com/mrlesmithjr/ansible-kea-dhcp/pull/9) that adds support for newer kea versions, but it does not work with Fedora 34 (I tested with moleule) since the wrong repo URL is produced. That PR does perhaps try to solve too many issues at once (e.g. support for Alpine and general change of formatting), and has not been merged.

jmidgren commented:
... I had a second look and it seems like ISC does not yet have (even version 1.9) packages for Fedora 34 - only some of the additional tools are being built yet. Fedora 33 packages are available though.

I also had another look at [the the other PR](https://github.com/mrlesmithjr/ansible-kea-dhcp/pull/9) again, and at second thought I think it solves the issue more in line with the original intention of the role (i.e. installing from ISC repos) - if only it worked for Fedora too :smiley: I managed to get it working for Fedora 33 with packages installed from the ISC repo doing only minor changes, which I verified by replacing the docker image pointed to by the `molecule.yml` file. 

My suggestion for this PR would be to add a mechanism for users to choose to install from distribution repo instead of from ISC repo (as I implied above). This could perhaps also lead to a cleaner separation of the "execution paths" in the task files, so instead of making exceptions for Fedora everywhere, handle packages installed from distro repos differently. Well, it's not up to me to decide of course, but I'm happy to help if you think this is a valid route.
